src/core/Akka/Dispatch/CachingConfig.cs

```diff
@@ -0,0 +1,277 @@
+﻿/**
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ * Original C# code written by Akka.NET project <http://getakka.net/>
+ */
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using Akka.Configuration;
+using Akka.Configuration.Hocon;
+
+namespace Akka.Dispatch
+{
+    /// <summary>
+    /// INTERNAL API
+    /// 
+    /// A <see cref="CachingConfig"/> is a <see cref="Config"/> that wraps another <see cref="Config"/> and is used to
+    /// cache path lookup and string retrieval, which we happen to do in some critical paths of the actor creation
+    /// and mailbox selection code.
+    /// 
+    /// All other <see cref="Config"/> operations are delegated to the wrapped <see cref="Config"/>.
+    /// </summary>
+    class CachingConfig : Config
```

Comment:
@rogeralsing truth be told, I'm not clear on how canonical Akka handles this. So I came up with my own solution.

Here's what I did:
Have the `Settings` class publish changes to `Dispatchers`
https://github.com/Aaronontheweb/akka.net/blob/dispatcher-squash-timestamp/src/core/Akka/Actor/Settings.cs#L27

And then have `Dispatchers` rebuild its `CachingConfig`:
https://github.com/Aaronontheweb/akka.net/blob/dispatcher-squash-timestamp/src/core/Akka/Dispatch/Dispatchers.cs#L167

This issue came up when I was integrating @Horusiath's latest Akka.Persistence changes. In all other cases when `InjectTopLevelFallback` is called it's done through the `ActorRefProvider` (i.e. in Akka.Remote / Akka.Cluster) but not in Akka.Persistence. So this is the solution I came up with to allow a plugin to inject its own dispatchers into the registry when it gets called.

The only other way of doing this AFAIK would be to scan all of the assemblies loaded into an Akka.NET AppDomain for embedded HOCON files (we would have to standardize the name) and chain all of them at the same time, rather than do it when they're invoked (plugins get invoked after `ActorRefProvider` and after `Dispatchers` is setup, which is what created this problem.)

I think that's what the JVM does and why all of their resource files are found in the same filename and path in each .jar.